In [1]:
import pandas as pd

In [2]:
dataset=pd.read_csv("PreprocessedPlacement.csv")

In [3]:
dataset

sl_no gender  ssc_p    ssc_b  hsc_p    hsc_b     hsc_s  degree_p  \
0      1.0      M  67.00   Others  91.00   Others  Commerce     58.00   
1      2.0      M  79.33  Central  78.33   Others   Science     77.48   
2      3.0      M  65.00  Central  68.00  Central      Arts     64.00   
3      4.0      M  56.00  Central  52.00  Central   Science     52.00   
4      5.0      M  85.80  Central  73.60  Central  Commerce     73.30   
..     ...    ...    ...      ...    ...      ...       ...       ...   
210  211.0      M  80.60   Others  82.00   Others  Commerce     77.60   
211  212.0      M  58.00   Others  60.00   Others   Science     72.00   
212  213.0      M  67.00   Others  67.00   Others  Commerce     73.00   
213  214.0      F  74.00   Others  66.00   Others  Commerce     58.00   
214  215.0      M  62.00  Central  58.00   Others   Science     53.00   

      degree_t workex  etest_p specialisation  mba_p      status    salary  
0     Sci&Tech     No     55.0         Mkt&HR  58.80      Placed  270000.0  
1     Sci&Tech    Yes     86.5        Mkt&Fin  66.28      Placed  200000.0  
2    Comm&Mgmt     No     75.0        Mkt&Fin  57.80      Placed  250000.0  
3     Sci&Tech     No     66.0         Mkt&HR  59.43  Not Placed       0.0  
4    Comm&Mgmt     No     96.8        Mkt&Fin  55.50      Placed       0.0  
..         ...    ...      ...            ...    ...         ...       ...  
210  Comm&Mgmt     No     91.0        Mkt&Fin  74.49      Placed       0.0  
211   Sci&Tech     No     74.0        Mkt&Fin  53.62      Placed  275000.0  
212  Comm&Mgmt    Yes     59.0        Mkt&Fin  69.72      Placed  295000.0  
213  Comm&Mgmt     No     70.0         Mkt&HR  60.23      Placed  204000.0  
214  Comm&Mgmt     No     89.0         Mkt&HR  60.22  Not Placed       0.0  

[215 rows x 15 columns]

In [5]:
dataset.dtypes

sl_no             float64
gender             object
ssc_p             float64
ssc_b              object
hsc_p             float64
hsc_b              object
hsc_s              object
degree_p          float64
degree_t           object
workex             object
etest_p           float64
specialisation     object
mba_p             float64
status             object
salary            float64
dtype: object

In [4]:
dataset.isna().sum()

sl_no             0
gender            0
ssc_p             0
ssc_b             0
hsc_p             0
hsc_b             0
hsc_s             0
degree_p          0
degree_t          0
workex            0
etest_p           0
specialisation    0
mba_p             0
status            0
salary            0
dtype: int64

In [6]:
dir(dataset) #find key word for covariance

['T',
 '_AXIS_ALIASES',
 '_AXIS_IALIASES',
 '_AXIS_LEN',
 '_AXIS_NAMES',
 '_AXIS_NUMBERS',
 '_AXIS_ORDERS',
 '_AXIS_REVERSED',
 '__abs__',
 '__add__',
 '__and__',
 '__array__',
 '__array_priority__',
 '__array_wrap__',
 '__bool__',
 '__class__',
 '__contains__',
 '__copy__',
 '__deepcopy__',
 '__delattr__',
 '__delitem__',
 '__dict__',
 '__dir__',
 '__div__',
 '__doc__',
 '__eq__',
 '__finalize__',
 '__floordiv__',
 '__format__',
 '__ge__',
 '__getattr__',
 '__getattribute__',
 '__getitem__',
 '__getstate__',
 '__gt__',
 '__hash__',
 '__iadd__',
 '__iand__',
 '__ifloordiv__',
 '__imod__',
 '__imul__',
 '__init__',
 '__init_subclass__',
 '__invert__',
 '__ior__',
 '__ipow__',
 '__isub__',
 '__iter__',
 '__itruediv__',
 '__ixor__',
 '__le__',
 '__len__',
 '__lt__',
 '__matmul__',
 '__mod__',
 '__module__',
 '__mul__',
 '__ne__',
 '__neg__',
 '__new__',
 '__nonzero__',
 '__or__',
 '__pos__',
 '__pow__',
 '__radd__',
 '__rand__',
 '__rdiv__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 

In [7]:
dataset.cov()

sl_no          ssc_p          hsc_p       degree_p  \
sl_no       3870.000000     -52.641355     -56.007056     -50.927664   
ssc_p        -52.641355     117.228377      47.626980      40.951272   
hsc_p        -56.007056      47.626980      90.478015      27.750339   
degree_p     -50.927664      40.951272      27.750339      51.303450   
etest_p       52.556168      37.659225      25.171634      23.422524   
mba_p          8.102336      24.535952      15.704758      16.403817   
salary   -228261.682243  661059.070637  425523.978700  339813.880026   

               etest_p        mba_p        salary  
sl_no        52.556168     8.102336 -2.282617e+05  
ssc_p        37.659225    24.535952  6.610591e+05  
hsc_p        25.171634    15.704758  4.255240e+05  
degree_p     23.422524    16.403817  3.398139e+05  
etest_p     176.251018    16.886973  8.485346e+04  
mba_p        16.886973    34.028376 -8.212396e+03  
salary    84853.458161 -8212.395566  1.768111e+10

In [8]:
dataset["degree_p"].mean()

66.25390697674419

In [11]:
dataset["degree_p"].sort_values()

14     50.0
25     50.2
91     50.8
34     52.0
49     52.0
       ... 
33     81.0
90     82.0
107    83.0
152    84.0
21     85.0
Name: degree_p, Length: 215, dtype: float64

In [13]:
dataset["etest_p"].sort_values()

20     50.00
190    50.00
99     50.00
19     50.48
32     50.89
       ...  
4      96.80
206    97.00
118    97.00
24     97.40
152    98.00
Name: etest_p, Length: 215, dtype: float64

#we already discused covariance means variance between two columns. And also the name bivariate shows two columns.
Assignment look 
degree_p and etest_p = 23.422 this value shows that the difference between two columns is 23.422
and its 'positive covariance' that means variance in small to large numers.

In [15]:
dataset.corr()

sl_no     ssc_p     hsc_p  degree_p   etest_p     mba_p    salary
sl_no     1.000000 -0.078155 -0.094649 -0.114294  0.063636  0.022327 -0.027595
ssc_p    -0.078155  1.000000  0.462450  0.528053  0.261993  0.388478  0.459165
hsc_p    -0.094649  0.462450  1.000000  0.407308  0.199331  0.283034  0.336432
degree_p -0.114294  0.528053  0.407308  1.000000  0.246317  0.392601  0.356790
etest_p   0.063636  0.261993  0.199331  0.246317  1.000000  0.218055  0.048067
mba_p     0.022327  0.388478  0.283034  0.392601  0.218055  1.000000 -0.010588
salary   -0.027595  0.459165  0.336432  0.356790  0.048067 -0.010588  1.000000

In this table we are going to find relation between two columns. 
salary and mba_p = -0.010588